src/Data/Record.purs

```diff
@@ -1 +1,113 @@
-module Data.Record where
+module Data.Record
+  ( get
+  , set
+  , modify
+  , insert
+  , delete
+  ) where
+
+import Data.Function.Uncurried (Fn2, Fn3, runFn2, runFn3)
+import Data.Symbol (class IsSymbol, SProxy, reflectSymbol)
+import Type.Row (class RowLacks)
+
+foreign import unsafeGet :: forall r a. Fn2 String (Record r) a
+foreign import unsafeSet :: forall r1 r2 a. Fn3 String a (Record r1) (Record r2)
+foreign import unsafeDelete :: forall r1 r2 a. Fn2 String (Record r1) (Record r2)
+
+-- | Get a property for a label which is specified using a value-level proxy for
+-- | a type-level string.
+-- |
+-- | For example:
+-- |
+-- | ```purescript
+-- | get (SProxy :: SProxy "x") :: forall r a. { x :: a | r } -> a
+-- | ```
+get
+  :: forall r r' l a
+   . IsSymbol l
+  => RowCons l a r' r
+  => SProxy l
+  -> Record r
+  -> a
+get l r = runFn2 unsafeGet (reflectSymbol l) r
+
+-- | Set a property for a label which is specified using a value-level proxy for
+-- | a type-level string.
+-- |
+-- | For example:
+-- |
+-- | ```purescript
+-- | set (SProxy :: SProxy "x")
+-- |   :: forall r a b. a -> { x :: b | r } -> { x :: a | r }
+-- | ```
+set
+  :: forall r1 r2 r l a b
+   . IsSymbol l
+  => RowCons l a r r1
+  => RowCons l b r r2
+  => SProxy l
+  -> b
+  -> Record r1
+  -> Record r2
+set l b r = runFn3 unsafeSet (reflectSymbol l) b r
+
+-- | Modify a property for a label which is specified using a value-level proxy for
+-- | a type-level string.
+-- |
+-- | For example:
+-- |
+-- | ```purescript
+-- | modify (SProxy :: SProxy "x")
+-- |   :: forall r a b. (a -> b) -> { x :: a | r } -> { x :: b | r }
+-- | ```
+modify
+  :: forall r1 r2 r l a b
+   . IsSymbol l
+  => RowCons l a r r1
+  => RowCons l b r r2
+  => SProxy l
+  -> (a -> b)
+  -> Record r1
+  -> Record r2
+modify l f r = set l (f (get l r)) r
+
+-- | Insert a new property for a label which is specified using a value-level proxy for
+-- | a type-level string.
+-- |
+-- | For example:
+-- |
+-- | ```purescript
+-- | insert (SProxy :: SProxy "x")
+-- |   :: forall r a. a -> { | r } -> { x :: a | r }
+-- | ```
+insert
+  :: forall r1 r2 l a
+   . IsSymbol l
+  => RowCons l a r1 r2
+  => SProxy l
+  -> a
+  -> Record r1
+  -> Record r2
+insert l a r = runFn3 unsafeSet (reflectSymbol l) a r
```

Comment:
This is what I've always thought. Since duplicate labels in records are nonsense, we should prevent them by construction. The problem is we have `Union` which will introduce duplicate labels. We would need a version that doesn't introduce duplicates.